### PR TITLE
 23.NEXT: Improvements (Fit Export, Tour Concatenation, Tour Blog)

### PR DESCRIPTION
**Pull request by FJBDev**
- Fit Export: Added the pause types
- Tour Blog: Improved the detection of http links in the tour description to make them clickable by using a [third party library](https://github.com/URL-Detector/URL-Detector) #1102
-  Tour Concatenation: The weather data is deleted from the concatenated tour

### CODE REVIEW

File: bundles/net.tourbook.common/src/net/tourbook/common/util/IAdvancedMenuForActions.java [hunk 1]
```diff
@@ -17,21 +17,21 @@
...
+   void onShowMenu();
 
-	/**
-	 * Resets data which are keept when the advanced menu is open
-	 */
-	void resetData();
+   /**
+    * Resets data which are kept when the advanced menu is open
+    */
+   void resetData();
```
Comment (wolfgang-ch) [inline bundles/net.tourbook.common/src/net/tourbook/common/util/IAdvancedMenuForActions.java:28]: When I do a cleanup of old code, then I also set the line delimiters to Unix

image

Before

image

After

image

File: bundles/net.tourbook.export/src/net/tourbook/export/fit/FitExporter.java [hunk 2]
```diff
@@ -237,8 +242,12 @@ private List<EventMesg> createEventMessages(final DateTime startTime,
             final EventMesg eventMesgStop = new EventMesg();
             final Date pausedTime_Start_Date = Date.from(Instant.ofEpochMilli(pausedTime_Start[index]));
             eventMesgStop.setTimestamp(new DateTime(pausedTime_Start_Date));
-            // By default: eventData == 1: auto-stop
-            eventMesgStop.setData(1L);
+            /**
+             * eventData == 0: user stop<br>
+             * eventData == 1: auto-stop
+             */
```
Comment (wolfgang-ch) [inline bundles/net.tourbook.export/src/net/tourbook/export/fit/FitExporter.java:248]: You copied my comment 😄 

image

**Review by wolfgang-ch — APPROVED** [review]

File: bundles/net.tourbook.export/src/net/tourbook/export/fit/FitExporter.java (hunk 2 above)
Comment (FJBDev) [reply bundles/net.tourbook.export/src/net/tourbook/export/fit/FitExporter.java:248]: Yes, it was well written so I didn't want to "reinvent the wheel" 😄

File: bundles/net.tourbook.common/src/net/tourbook/common/util/IAdvancedMenuForActions.java (hunk 1 above)
Comment (FJBDev) [reply bundles/net.tourbook.common/src/net/tourbook/common/util/IAdvancedMenuForActions.java:28]: Good to know. Is it possible that this is automatically applied when saving a file like other actions ?

Comment (wolfgang-ch) [reply bundles/net.tourbook.common/src/net/tourbook/common/util/IAdvancedMenuForActions.java:28]: I've not seen your requested feature in the 122 save actions

**Pull request merged**